### PR TITLE
test: modify TestServerMetricsReporting

### PR DESCRIPTION
**Pull request by miparnisari**
## Description
Closes https://github.com/openfga/openfga/issues/1076

### CODE REVIEW

File: pkg/storage/mysql/mysql.go [hunk 1]
```diff
@@ -32,6 +32,7 @@ type MySQL struct {
 	stbl                   sq.StatementBuilderType
 	db                     *sql.DB
 	logger                 logger.Logger
+	dbStatsCollector       prometheus.Collector
```
Comment (jon-whit) [inline pkg/storage/mysql/mysql.go:35]: Why add the collector explicitly here instead of just using the default collector registration?

**Review by jpadilla — APPROVED** [review]

**Review by jon-whit — CHANGES_REQUESTED** [review]
What is the problem in https://github.com/openfga/openfga/issues/1076? It's not clear what the objective is with this PR? The reason I ask is that I strongly discourage us from explicitly passing around collectors unless we have a very strong reason to not use default collector registration.

**Comment by miparnisari** [comment]
> What is the problem in https://github.com/openfga/openfga/issues/1076?

It works but we don't have proper tests for it 
image

I wouldn't want to introduce a regression in the publishing of metrics because it'd lead to a gap of valuable data.

> It's not clear what the objective is with this PR? The reason I ask is that I strongly discourage us from explicitly passing around collectors unless we have a very strong reason to not use default collector registration.

We use the default collector registration. But for the tests I needed to properly de-register once done using them because otherwise i would get "attempt to register duplicate collector"

File: pkg/storage/mysql/mysql.go (hunk 1 above)
Comment (miparnisari) [reply pkg/storage/mysql/mysql.go:35]: We use the default, but I need to deregister after closing to have the tests work properly

File: cmd/run/run.go [hunk 2]
```diff
@@ -465,8 +465,9 @@ func (s *ServerContext) Run(ctx context.Context, config *serverconfig.Config) er
 		s.Logger.Info(fmt.Sprintf("📈 starting metrics server on '%s'", config.Metrics.Addr))
 
 		go func() {
-			http.Handle("/metrics", promhttp.Handler())
-			if err := http.ListenAndServe(config.Metrics.Addr, nil); err != nil {
+			mux := http.NewServeMux()
+			mux.Handle("/metrics", promhttp.Handler())
+			if err := http.ListenAndServe(config.Metrics.Addr, mux); err != nil {
```
Comment (miparnisari) [inline cmd/run/run.go:470]: This change prevents

```
2023-10-27T11:07:41.557-0700	INFO	🚀 starting openfga service...	{"version": "dev", "date": "unknown", "commit": "none", "go-version": "go1.21.3"}
panic: http: multiple registrations for /metrics

goroutine 447 [running]:
net/http.(*ServeMux).Handle(0x31cac80, {0x25d3d41, 0x8}, {0x27fd440?, 0xc0005632f0})
	/usr/local/opt/go/libexec/src/net/http/server.go:2530 +0x3bd
net/http.Handle(...)
	/usr/local/opt/go/libexec/src/net/http/server.go:2573
github.com/openfga/openfga/cmd/run.(*ServerContext).Run.func2()
	/Users/mparnisari/GitHub/openfga/cmd/run/run.go:468 +0x88
created by github.com/openfga/openfga/cmd/run.(*ServerContext).Run in goroutine 440
	/Users/mparnisari/GitHub/openfga/cmd/run/run.go:467 +0x33c5
```

Comment (jon-whit) [reply cmd/run/run.go:470]: Taking a look. There used to be a good prometheus test package to de-register between tests

**Review by jon-whit — APPROVED** [review]

**Pull request merged**